File: src/VB.NET/MetadataWebApi/MetadataWebApi/DataFile.vb
' -----------------------------------------------------------------------
'  <copyright file="DataFile.vb" company="Experian Data Quality">
'   Copyright (c) Experian. All rights reserved.
'  </copyright>
' -----------------------------------------------------------------------

Imports System.Runtime.Serialization

''' <summary>
''' A class representing a single data file.
''' </summary>
<DataContract(Namespace:="", Name:="DataFile")> _
<DebuggerDisplay("{FileName} {MD5Hash}")> _
Public Class DataFile

    ''' <summary>
    ''' Gets or sets the file name.
    ''' </summary>
    <DataMember(Name:="FileName")> _
    Public Property FileName As String

    ''' <summary>
    ''' Gets or sets the MD5 hash of the file.
    ''' </summary>    
    <DataMember(Name:="Md5Hash")> _
    Public Property MD5Hash As String

    ''' <summary>
    ''' Gets or sets the size of the file in bytes.
    ''' </summary>
    <DataMember(Name:="Size")> _
    Public Property Size As Long

End Class

File: src/CSharp/MetadataWebApi/MetadataWebApi/DataFile.cs
//-----------------------------------------------------------------------
// <copyright file="DataFile.cs" company="Experian Data Quality">
//   Copyright (c) Experian. All rights reserved.
// </copyright>
//-----------------------------------------------------------------------

using System.Diagnostics;
using Newtonsoft.Json;

namespace Experian.Qas.Updates.Metadata.WebApi.V1
{
    /// <summary>
    /// A class representing a single data file.
    /// </summary>
    [DebuggerDisplay("{FileName} {MD5Hash}")]
    public class DataFile
    {
        /// <summary>
        /// Gets or sets the file name.
        /// </summary>
        [JsonProperty("FileName")]
        public string FileName { get; set; }

        /// <summary>
        /// Gets or sets the size of the file in bytes.
        /// </summary>
        [JsonProperty("Size")]
        public long Size { get; set; }

        /// <summary>
        /// Gets or sets the MD5 hash of the file.
        /// </summary>
        [JsonProperty("Md5Hash")]
        public string MD5Hash { get; set; }
    }
}
